create manifesto.md

zhekson1 opened the pull request.

zhekson1 committed:
Update manifesto.md

zhekson1 committed:
Update manifesto.md

zhekson1 committed:
Update manifesto.md

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+6. Future Directions
+7. Conclusion & Support
+
+## 1 - Abstract
```
The abstract doesn't mention expressivity at all. If the contemporary DApps became composable, you still wouldn't have a real economy unless users are able to express their market preferences without having to "get permission" from the DApp creators. For example, Liqwid requires a multi-sig from the creators in order to add support for a new asset or collateral. Basically, if a new nation was created and issued its own currency, it wouldn't be able to participate in the DeFi economy without explicit approval from the multisig. Also DEXs that uses algorithms to determine the price will never be able to accurately represent true market sentiment. This can only cause problems for DeFi.

A blockchain economy will never be mass adopted as long as this is the status quo. Impermanent loss is literally a symptom of the DApps not being able to express the true sentiment of users. In my view, the lack of permissionless expressivity is actually the biggest impediment to mass adoption.

EDIT: Commenting here again after reading the rest of it. To me, the actual principles are (in order of importance):
1. Maximally expressive - expressiveness is the number one requirement for price discovery, proper risk management, and a secure PoS blockchain. Even composability actually falls under the umbrella of expressivity. If you can't express that you only want to buy this bond if you are also guaranteed to get the next payment, you can't properly manage risk. If users can vote with their current assets (ie, use version 4 instead of version 3 and delegate to pool A instead of B), there is no reason to need for governance tokens at all. All of the current ailments of DeFi are a direct result of the developers not prioritizing expressivity.
3. Availability
4. Scalability

eUTxO lets us have all three at the same time.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+----
+WIP
+
+Cardano's eUTxO model, with its asset-oriented programming paradigm, is uniquely well suited to achieve this balance. 
```
I'm not sure what "asset-oriented programming paradigm" means. I've never seen it used outside of Radix.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+
+Cardano's eUTxO model, with its asset-oriented programming paradigm, is uniquely well suited to achieve this balance. 
+
+This is especially true in the realm of DeFi, where eUTxO eases development of robust, scalable, and capital efficient protocols. However, its full potential has yet to be realized, largely due to its novelty, and thus a lack of standards. 
```
Expressivity should also be mentioned as a key result of the eUTxO. Since everyone gets their own UTxOs, those UTxOs can be completely personalized. Capital efficiency is a consequence of this expressivity.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+### 3.4 - Questionable Sustainability
+Sustainability is key to the viability of an alternative financial system. Many DeFi protocols have an associated "utility" token, which carries heavy implications for the sustainability of the protocol. Below are discussions of a few areas where such tokens' sustainability is questionable, at best.
+
+#### Capital Inefficiency
```
Again capital efficiency is determined by the level of expressivity. Impermanent loss only exists because DApps don't allow users to set their own price ranges. Fractionalized liquidity pools, like what Genius yield uses, are better but still not ideal. Order-books are the only architecture that allows for full expressivity which is why it is more capital efficient.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+The popularity of Accounts based ledgers gave rise to the pooled AMM-style architectures that dominate the DeFi landscape today. However, AMM liquidity providers suffer from inventory risk (i.e. impermanent loss), due to the models' inefficient provisioning of liquidity over the cost curve. Even for AMMs with "concentrated liquidity", such models can never be as capital efficient as a true orderb ...
+
+#### Constrained Lending/Borrowing
+The above point (capital inefficiency) applies in a similar manner to lending/borrowing protocols. This is perhaps even more impactful, since credit/debt markets are usually the largest market(s) in a mature economy. Many lending/borrowing protocols on Cardano employ pooled liquidity architectures, which necessitates the constraining of loan terms, thereby limiting lenders' flexibility to issue l ...
```
Here too, the actual issue is lack of expressivity. The free market should decide everything.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+Semantics aside, the point is that **collectively-managed lending/borrowing protocols are not an issue in and of themselves, but they cannot be the *only* option in a robust economy.**
+
+#### Yield Farming - WIP
+Many DeFi protocols incentivize liquidity via "yield-farming", where inventory (or liquidation/collateralization) risk is compensated with a token that, optimistically, derives its value from profit-sharing with the protocol's market makers. This means the protocol must charge "batcher" fees *in addition* to regular Cardano transaction fees, which limits the viability of smaller-sized trades. Usu ...
```
This section seems like the only section that actually pertains to the sustainability of "utility" tokens topic.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+### 3.3 - Threats to Ouroboros PoS
+Many DeFi protocols sacrifice full delegation/voting control, especially in cases of "pooled" liquidity. Such protocols usually attempt to compensate via DAO-based delegations, but this is never as expressive individual stake-control, and opens the door to a variety of governance-based attacks. There are too many scenarios to list here, but suffice to say that **the more delegation control is dec ...
+
+### 3.4 - Questionable Sustainability
```
I think this section should be broken up into "Lack of Expressivity" and "Questionable Sustainability".

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+1. Who decides who gets to be a batcher or oracle, and how are they compensated?
+2. If the answer to #1 is DAOs/dApp-tokens, what guardrails are in place to prevent batchers/oracles from using their profits to gradually become their own overseers?
+
+To be clear, good governance is not impossible, but increases in difficulty proportional to the width of a protocol's scope. The scope of something like DeFi (the basis of an alternative financial system) is so wide and the attack vector is so large that governance should be minimized, wherever possible. Hence, there is a need for agreed upon standards in the DeFi space, at least within the L1 co ...
```
I think you should mention that, if expressivity was actually prioritized, there would be no need for governance tokens at all. The whole thing is just a bandaid on a bad design decision.

zhekson1 replied:
"solutions" such as this will be in the following section, here I am laying out the deficiencies. can briefly touch on it though

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,321 @@
+
+
+## 4 - A Return to First Principles
+Astute readers will have noticed a pattern in the previous section; each subsection's critique centers around one or both of the following two points:
```
And lack of expressivity.

zhekson1 replied:
lack of expressivity is a consequence of these design choices; they are the root cause

zhekson1 committed:
Update section 3 and start new intro section

zhekson1 committed:
intro to the intro and funding sections

fallen-icarus commented on manifesto.md:
```diff
@@ -49,10 +49,23 @@ This is especially true in the realm of DeFi, where eUTxO eases development of r
+Again, the purpose of money (and money systems) is to propagate information that reflects underlying *reality* as faithfully and with as little friction as possible. Since this reality is nothing more than the collective value-differences of individuals, and since swift allocation of resources depends on individuals' ability to *always express* these differences, then **the most effective money s ...
+
+#### Historical Precedent
+From bartering, to commodity-moneys, to commodity-backed paper, to derivatives and modern financial institutions; technological innovation fuels this trend, pushing it towards the aforementioned ideal in a darwinian manner. For example, although paper notes are obviously a faster and more expressive method of information transfer than raw commodities, it wasn't until the rise of an entity or syst ...
```
I think you should tie this paragraph more closely to the last two sentences of the previous section. For example, despite the efficiencies and speed of using dutch fiat being available, any areas where the dutch empire could not defend the reliability of the notes devolved back to using commodities. This is direct evidence that scalability is less important than reliability. The is true everywhere throughout history. Societies have always chosen to sacrifice speed of a money system in favor of resilience. They only "upgrade" the speed when there is no cost to resilience. 

The above is sort of implied in the paragraph, but given how critical it is to the whole manifesto, I think the connection should be made explicit by giving the example of what happens outside of the network.

fallen-icarus commented on manifesto.md:
```diff
@@ -267,15 +280,25 @@ Cardano would not be where it is without them, and needs them to keep evolving a
 
-In the near to mid term, this project will require an enormous grassroots lift. There is nothing to sell, no token/equity for would-be investors, and a lot of work to do. 
+- Delegations are sticky; once a delegation is made, it sticks around until the delegator chooses to opt out. This equates to a more reliable/consistent income stream, as opposed to the sporadic nature of donations. 
+- Projects that opt for token-based fundraising (i.e. public sales, ISPOs, e.t.c.) are subject to regulatory scrutiny, especially if their project is successful enough that it poses a threat to governments or the financial elite. On the other hand, the nature of the SPO-delegator relationship in Cardano PoS lends itself to less regulatory scrutiny, since there is never anything that could be cons ...
```
I'd also compare it to catalyst. 
- Catalyst requires developers to seek re-funding every year which diverts energy away from development and may create short term incentives (appease would-be voters at the expense of long-term trade-offs).
- Delegators can undelegate at any time if they think the project is going in the wrong direction. Since in order to do this, delegators will want to watch progress, this actually encourages open-source development more so than catalyst.
- Catalyst requires milestone based funding with estimated deadlines but in reality, it is very hard to estimate how long something will take and what it will ultimately look like. If the final result is very different than what was promised in catalyst, you don't get the money despite the result possibly being a better market fit. This means catalyst isn't properly suited to true agile R&D while delegation is.

zhekson1 committed:
Update manifesto.md

zhekson1 committed:
Update manifesto.md

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Again, the purpose of money (and money systems) is to propagate information that reflects underlying *reality* as faithfully and with as little friction as possible. Since this reality is nothing more than the collective value-differences of individuals, and since swift allocation of resources depends on individuals' ability to *always express* these differences, then **the most effective money s ...
+
+#### Historical Precedent
+From bartering, to commodity-moneys, to commodity-backed paper, to derivatives and modern financial institutions; technological innovation fuels this trend, pushing it towards the aforementioned ideal in a darwinian manner. For example, although paper notes are obviously a faster and more expressive method of information transfer than raw commodities, it wasn't until the rise of an entity or syst ...
```
I don't think this paragraph fully addresses the money vs commodity debate. The last part of the previous section is:
> Order is important here; speed of propagation is a competitive advantage for a money system, but speed cannot under any circumstances undermine the system's reliability nor its accuracy, lest the misallocation of resources. This is not purely theoretical; it is evidenced by the evolution of money throughout history.

but this paragraph doesn't explain it. You sort of dance around it. Money can easily be printed which can dramatically distort the value-network. The point of the admin network is to defend against exactly this attack vector. It is astronomically harder to increase the supply of commodities like gold which is why gold can be used even without an admin network. This attack vector doesn't impact gold.

Money is much more scalable than commodities but the reliability of the value-network is more important. History has shown time and time again that whenever the admin network can no longer properly control the money supply (and therefore the value-network), societies **always** revert back to using gold. They would rather sacrifice throughput than the reliability of the value-network. 

This argument needs to be made explicit.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Note:  Every protocol doesn't suffer from *all* of these properties, but enough of them do, so as to underrepresent the power of Cardano-based DeFi. 
+
+#### 3.1 - Fractured, Non-Composable Protocols
+One of the greatest advantages of eUTxO is *atomic composability*, or the ability to "combine" different pieces of (related or unrelated) logic in the same transaction. This is great for throughput, but also enables the creation of complex, multi-step protocols. However, composability is not a given; it only works for contracts that are co-designed to work well with each other. Here lies one of t ...
```
There is a belief that batcher based dapps can compose so I think it is worth stressing that this is false. Arbitrary composition cannot be trustlessly enforced if batchers are required. Transactions simply don't have the room to support logic for arbitrary composition **in addition to** the actual logic for the dapps being composed. This logic would need to be in a meta smart contract. It also begs the question if a meta smart contract can even properly enforce all arbitrary compositions, both for current protocols and future protocols; can these arbitrary compositions even be codified in a smart contract?

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Many DeFi protocols sacrifice full delegation/voting control, especially in cases of "pooled" liquidity. Such protocols usually attempt to compensate via DAO-based delegations, but this is never as expressive individual stake-control, and opens the door to a variety of governance-based attacks. There are too many scenarios to list here, but suffice to say that **the more delegation control is dec ...
+
+#### 3.5 Low Expressivity
+The viability of *any* financial system depends on its constituency's freedom to express individual sentiment. This is a foundational principle of free market economies; any deviation from individuals' ability to express themselves (financially) distorts the information system that is money and prices. The extent of this distortion is nonlinear and difficult to predict, so minimizing it is of cri ...
```
This section also needs to explain risk-management. An economy is all about taking risk (like starting businesses). The easier it is to manage risk, the more complex an economy can form. Risk management is directly proportional to how expressive the system is. An economy that hinders risk management will never grow very large. In a way, risk management is like the base of an economic pyramid: the easier it is to manage risk, the wider the foundation can be and the bigger the overall pyramid can be.

The below two sub-sections are due to risk-management issues from a lack of expressivity.

fallen-icarus replied:
Risk management is actually the reason why order-books are fundamentally better than liquidity pools. Since you can exactly express your desired price, you can set how much you are willing to lose. You don't have this control with liquidity pools. Impermanent loss is a symptom of a lack of proper risk management options.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+##### 3.5.2 Constrained Lending/Borrowing
+The above point (capital inefficiency) applies in a similar manner to lending/borrowing protocols. This is perhaps even more impactful, since credit/debt markets are usually the largest market(s) in a mature economy. Many lending/borrowing protocols on Cardano employ pooled liquidity architectures, which necessitates the constraining of loan terms, thereby limiting lenders' flexibility to issue l ...
+
+Semantics aside, the point is that **collectively-managed lending/borrowing protocols are not an issue in and of themselves, but they cannot be the *only* option in a robust economy.**
```
You keep saying this but I don't think it fully captures the seriousness of the situation. I think the following is more accurate:

> Semantics aside, the point is that **collectively-managed lending/borrowing protocols are not an issue in and of themselves, but they cannot be the *foundation* of a robust economy.**

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Blockchains are a similar beast - as the technology matures, the number of possible applications (even within one vertical, like DeFi) explodes. Currently, every DeFi provider architects an entire stack, from smart contracts to the frontend, each of which is siloed and lacks interoperability with others. 
+
+
+
```
This section needs to also explain why the foundation of DeFi **must** be on L1.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+## 1 - Abstract
+Blockchain-based decentralized finance (DeFi) protocols have long sought to provide an alternative financial platform for the disenfranchised or unbanked peoples of the world. Although we've come a long way since Bitcoin's genesis, the industry has yet to deliver on such promises sustainably, at scale. A key reason for this is the lack of a comprehensive and unified framework for DeFi, even withi ...
+
+1. The full potential of the eUTxO model cannot be realized without a set of maximally expressive (and composable) protocols, upon which domain-specific scaling solutions are built. 
```
The abstract should be mostly chain agnostic. Most of it should be about how DeFi requires the first principles and how, across all chains, DApps are not being designed properly. Then you can mention how cardano is uniquely suited for DeFi but cardano DApps are also not properly designed.

The current abstract focuses too much on cardano while the manifesto is true for all chains.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+From bartering, to commodity-moneys, to commodity-backed paper, to derivatives and modern financial institutions; technological innovation fuels this trend, pushing it towards the aforementioned ideal in a darwinian manner. For example, although paper notes are obviously a faster and more expressive method of information transfer than raw commodities, it wasn't until the rise of an entity or syst ...
+
+#### Issues with Current World Order
+Ever since the end of WW2, and especially after the collapse of the Soviet Union, the United States has been the dominant economic and military power of the world. Consequently, the Dollar system has been the most reliable avenue for individuals to express their value-differences. However, the dollar system is not without its problems:
```
I think this section should be reworked. The main idea should be that all moneys eventually die due to either a weakening internal economy that weakens the admin (triffin's dilemma) or discretionary abuse; it is usually a combination of the two. Blockchain offers a way to have money that is immune to both.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+<POOLED ARCHITECTURE TOO?>
+
+#### 3.3 - Low Availability Guarantees
+An alternative financial system will likely be perceived as a threat by those in positions of high (legacy) power, who may employ a variety of regulatory or cyber-offensive tactics to disrupt it. Current DeFi protocols on Cardano rely on "enshrined" off-chain entities (i.e. oracles, batchers, e.t.c.), which greatly increases the protocols' attack surface. If DeFi is to ever meet the needs of a gl ...
```
I don't think this argument is correct: just because a protocol uses batchers doesn't mean it has low availability. Cardano uses batchers (aka stake pools). Permissioned batchers create low availability since they are easy to attack but permissionless batchers (just like stake pools) do not. The same is true for oracles. The real issue with batchers is the lack of trustless expressivity. Oracles don't actually have an intrinsic availability issue; they just can't help create an independent economy on cardano. A true defi economy should be mostly endogenous (perhaps this should also be a first principle?); the more it relies on an external economy that is under the influence of a government, the more that government can influence the defi economy.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+#### 3.5 Low Expressivity
+The viability of *any* financial system depends on its constituency's freedom to express individual sentiment. This is a foundational principle of free market economies; any deviation from individuals' ability to express themselves (financially) distorts the information system that is money and prices. The extent of this distortion is nonlinear and difficult to predict, so minimizing it is of cri ...
+
+To be clear, concentrating liquidity into collectively managed pools is not an issue in and of itself - in many cases such entities serve to accelerate price discovery and market efficiency (i.e. hedge funds, banks, .e.t.c.). The issue is when such entities are the *only* avenue for participating in the DeFi economy, in which case their outsized influence on the market undermines efficient price  ...
```
This needs to be reworded to stress that managed pools can't be the foundation. You also need to mention it is fundamentally impossible for an algorithm based solely on supply and demand to accurately reflect market sentiments which also depends on intangible things (eg, personal risk tolerance, macro risks, etc). Intangible things can't be seen by the blockchain, even with oracles which means the algorithm will always be less accurate than if users just specified prices directly. This is absolutely critical for healthy allocation of resources.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+To be clear, concentrating liquidity into collectively managed pools is not an issue in and of itself - in many cases such entities serve to accelerate price discovery and market efficiency (i.e. hedge funds, banks, .e.t.c.). The issue is when such entities are the *only* avenue for participating in the DeFi economy, in which case their outsized influence on the market undermines efficient price  ...
+
+##### 3.5.1 Capital Inefficiency
+The popularity of Accounts based ledgers gave rise to the pooled AMM-style architectures that dominate the DeFi landscape today. However, AMM liquidity providers suffer from inventory risk (i.e. impermanent loss), due to the models' inefficient provisioning of liquidity over the cost curve. Even for AMMs with "concentrated liquidity", such models can never be as capital efficient as a true orderb ...
```
Tbh, I don't like how technical this paragraph is. It seems needlessly technical. The points are:
1. An algorithm fundamentally cannot accurately price assets, even with oracles.
2. There is too little risk management, even with fractionalized liquidity pools.

Nothing else matters. The above applies to all chains, regardless of Account or eUTxO. Pooled liquidity simply cannot serve as the foundation on *any* chain for the above reasons.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+The popularity of Accounts based ledgers gave rise to the pooled AMM-style architectures that dominate the DeFi landscape today. However, AMM liquidity providers suffer from inventory risk (i.e. impermanent loss), due to the models' inefficient provisioning of liquidity over the cost curve. Even for AMMs with "concentrated liquidity", such models can never be as capital efficient as a true orderb ...
+
+##### 3.5.2 Constrained Lending/Borrowing
+The above point (capital inefficiency) applies in a similar manner to lending/borrowing protocols. This is perhaps even more impactful, since credit/debt markets are usually the largest market(s) in a mature economy. Many lending/borrowing protocols on Cardano employ pooled liquidity architectures, which necessitates the constraining of loan terms, thereby limiting lenders' flexibility to issue l ...
```
> Many lending/borrowing protocols on Cardano employ pooled liquidity architectures, which necessitates the constraining of loan terms, thereby limiting lenders' flexibility to issue loans as they best see fit.

I think this needs to be explained. You just make the statement and move on. Collective decision making fundamentally means some people will always be disappointed because their preferences were not chosen. Any kind of collective decision making shouldn't be used to determine prices; it doesn't matter if it is algorithmic based or DAO based.

> In many cases this also means assets must go through some kind of centralized or DAO-based "approval" process before they can be lent/borrowed.

I think you should give the example of a new nation being created that looks to add its new currency to the defi ecosystem. Any kind of approval process (centralized or DAO based) means joining the economy is fundamentally permissioned. This is awful since it limits how easy it is for nations to compete economically.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+
+Firstly, initial liquidity is often catalyzed by an aggressive emission schedule (rapidly inflates supply & decreases its value), which is obviously unsustainable. This is justified as a "bootstrapping" phase, followed by some notion of a path towards sustainable value generation - usually in the form of profit sharing with the protocol's market makers. This means the protocol itself must somehow ...
+
+1. Some protocols collect "batcher" fees that are split between batcher providers and utility token holders. However, such fees are charged *on top* of regular Cardano transaction fees, which limits the viability of smaller-sized trades. Furthermore, it means such protocols suffer from all the drawbacks of relying on "enshrined" off-chain entities, as mentioned in previous sections. This ultimate ...
```
I think you should add that, in some cases, the batcher fees result in fees that are 10x higher than other protocols. This helps hammer home the uncompetitiveness of this approach.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Firstly, initial liquidity is often catalyzed by an aggressive emission schedule (rapidly inflates supply & decreases its value), which is obviously unsustainable. This is justified as a "bootstrapping" phase, followed by some notion of a path towards sustainable value generation - usually in the form of profit sharing with the protocol's market makers. This means the protocol itself must somehow ...
+
+1. Some protocols collect "batcher" fees that are split between batcher providers and utility token holders. However, such fees are charged *on top* of regular Cardano transaction fees, which limits the viability of smaller-sized trades. Furthermore, it means such protocols suffer from all the drawbacks of relying on "enshrined" off-chain entities, as mentioned in previous sections. This ultimate ...
+2. Some protocols create value for their tokens by sharing profits generated by liquidity providers / market makers. However, collective liquidity provisioning is only necessary for pooled DeFi architectures, which are inferior to orderbooks for the reasons mentioned in previous sections. And since orderbooks *don't* require collective liquidity and will outcompete AMMs, profit sharing cannot sus ...
```
I am not sure this point is true. MMs can work on the peripheral where they settle against the order book. But this requires an element of active trading and hedging. Collectively run automated MMs are the problem.
1) Collective decision making is too slow to react to market changes (like if an algorithm must be updated).
2) An algorithm fundamentally cannot accurately reflect reality to the point where it can just run forever. It must always be "guided" by humans. Who will these privileged humans be?

The profit sharing isn't what makes the AMMs unsustainable; it's the automation that makes it unsustainable at large scales. An economy cannot be run by an algorithm; it will never be anywhere near as accurate as a bottom-up price discovery prices like order-books. This is why AMMs can't be the *foundation* of a DeFi economy.

zhekson1 replied:
gotta discuss this point further

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+1. Some protocols collect "batcher" fees that are split between batcher providers and utility token holders. However, such fees are charged *on top* of regular Cardano transaction fees, which limits the viability of smaller-sized trades. Furthermore, it means such protocols suffer from all the drawbacks of relying on "enshrined" off-chain entities, as mentioned in previous sections. This ultimate ...
+2. Some protocols create value for their tokens by sharing profits generated by liquidity providers / market makers. However, collective liquidity provisioning is only necessary for pooled DeFi architectures, which are inferior to orderbooks for the reasons mentioned in previous sections. And since orderbooks *don't* require collective liquidity and will outcompete AMMs, profit sharing cannot sus ...
+
+Hence, we are left with a situation where users have no choice but to use protocols whose viability depends not on technical merit alone, but on the value of the corresponding token. And since the parameters of such protocols (especially those to do with liquidity) are collectively managed, users' financial expressivity is limited, which impedes market efficiency. As soon as a more efficient prot ...
```
This paragraph needs to stress that liquidity pools are only temporarily stable due to the aforementioned points. They are not capable of lasting for decades if not centuries like order-books.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Hence, we are left with a situation where users have no choice but to use protocols whose viability depends not on technical merit alone, but on the value of the corresponding token. And since the parameters of such protocols (especially those to do with liquidity) are collectively managed, users' financial expressivity is limited, which impedes market efficiency. As soon as a more efficient prot ...
+
+##### Governance Risk
+Protocols that opt for token/plutocratic-based governance are subject to a pandora's box of concerns, especially when "enshrined" actors (oracles, batchers e.t.c.) are involved. Present difficulties/debates over Cardano's governance are a testament to how hard this problem is. Governance is a complex topic beyond the scope of this document, so, for the sake of brevity, consider the following two  ...
```
I don't think this section is correct. Batchers can be permissionless and paid in ADA which means neither of your 2 points apply. The real issue with governance is that it is fake: just because a vote goes one way doesn't mean the protocol will trustlessly update in that direction. It still requires the developers to actually implement the change and these developers can refuse or go in a different direction.

Also, if governance tokens are issued, this is a major regulatory obstacle if the issuance is not 100% decentralized. Plus, these governance tokens are almost never distributed in a way to have a diverse government which means the whole dapp is now unstable.

fallen-icarus replied:
This section should probably also go before the yield farming section since part of the issue with AMMs revolve around collective decision making.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+To be clear, good governance is not impossible, but increases in difficulty proportional to the width of a protocol's scope. The scope of something like DeFi (the basis of an alternative financial system) is so wide and the attack vector is so large that governance should be minimized, wherever possible. Hence, there is a need for agreed upon standards in the DeFi space, at least within the L1 co ...
+
+##### Regulatory Concerns
+Protocol-specific "utility" tokens impose high regulatory burdens or uncertainties to users and developers, particularly with regard to securities laws. As mentioned in the previous section, the value of the token is usually tightly coupled with a single team or product, which can easily illicit action from regulators. In fact, this may be a good litmus test for distinguishing "products" from "pr ...
```
> ... the value of the token is usually tightly coupled with a single team or product, which can easily illicit action from regulators.

I get what you are trying to say here but I don't think it works. Technically, I am a single team and the p2p protocols are my product. I think the real test is what happens if a team walks away? Does the protocol development stop forever? Does the protocol stop running? On cardano, if the team walked away, what happens to the issuance of the utility tokens?

If the minting of the utility or governance tokens is not 100% decentralized, what is stopping a government from getting access to the relevant smart contracts? If a malicious government got a hold of the minting policy for the utility tokens, they can basically create a "FED" for that DApp. If they got the minting policy for the governance tokens, they can mint enough new tokens so that they control the DApp entirely; this either kills the DApp (as people leave to go elsewhere) or the government now monopolizes the DApp. In short, if the tokens are not 100% decentralized, they actually weaken the availability of the DApp.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,288 @@
+Semantics aside, the point is that **collectively-managed lending/borrowing protocols are not an issue in and of themselves, but they cannot be the *only* option in a robust economy.**
+
+#### 3.5 - Questionable Sustainability
+Sustainability is key to the viability of an alternative financial system. Many DeFi protocols have an associated "utility" token, which carries heavy implications for the sustainability of the protocol. Below are discussions of a few areas where such tokens' sustainability is questionable, at best.
```
You don't mention funding of protocol development anywhere here. Issuing tokens to fund development is a terrible idea due to the regulatory risks. Stake pools are better.

The number one issue critique for projects that don't issue tokens is funding but this is 100% false on cardano.

zhekson1 replied:
not sure if this belongs in a sustainability critique of defi as a whole, as some token issuance (even security-like tokens) makes sense for some protocols/products. I can make the distinction, but I feel like its going overboard considering we mention it in our funding section.

fallen-icarus replied:
The point is that there is widespread belief that you can't fund development without tokens. This manifesto needs to debunk that myth. I agree that there are scenarios where the token is important but for the vast majority of use cases, stake pools are better.

zhekson1 committed:
Update manifesto.md

zhekson1 committed:
Update manifesto.md

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,287 @@
+#### Historical Precedent
+From bartering, to commodity-moneys, to commodity-backed paper, to derivatives and modern financial institutions; technological innovation fuels this trend, pushing it towards the aforementioned ideal in a darwinian manner. For example, although paper notes are obviously a faster and more expressive method of information transfer than raw commodities, it wasn't until the rise of an entity or syst ...
+
+And so we have the long arc of an Empire's money system, or what Ray Dalio calls the "Long Term Debt Cycle". The main idea is that all fiat moneys eventually decay due to weakening socioeconomics, which undermines the Empire's power to reliably administer the network. For small economies this usually leads to outright default, whereas in large economies (whose money is used as a global reserve) t ...
```
Discretionary abuse is not the same thing as triffin's dilemma. Triffin's dilemma outlines the fundamental requirements of what needs to happen for a nation's currency to be a global currency. Those requirements force the nation to go against its own best interests.

DLT based currencies offer a way to have a global currency that is not impacted by triffin's dilemma. This point is way too important to just give it a few sentences.

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,287 @@
+Again, the purpose of money (and money systems) is to propagate information that reflects underlying *reality* as faithfully and with as little friction as possible. Since this reality is nothing more than the collective value-differences of individuals, and since swift allocation of resources depends on individuals' ability to *always express* these differences, then **the most effective money s ...
+
+#### Historical Precedent
+From bartering, to commodity-moneys, to commodity-backed paper, to derivatives and modern financial institutions; technological innovation fuels this trend, pushing it towards the aforementioned ideal in a darwinian manner. For example, although paper notes are obviously a faster and more expressive method of information transfer than raw commodities, it wasn't until the rise of an entity or syst ...
```
This paragraph is going to lose some readers by being only abstract. You need to also give a concrete example to illustrate that fiat moneys need reliable admin networks to control its supply while commodities like gold do not. When I say make this explicit, I mean literally say this concretely. Not abstractly.

fallen-icarus replied:
It can literally be a sentence or two but giving the concrete example helps readers make the connections.

zhekson1 committed:
Update principles section

fallen-icarus commented on manifesto.md:
```diff
@@ -1 +1,297 @@
+##### Resilience = Availability + Sustainability
+Above all else, value-information networks must be reliable and resilient against disruption. This is necessary to outcompete the incumbent system (by being the preferred choice for most users) while simultaneously thwarting disruption by that same incumbency (who are incentivized to do so). This imposes two constraints/requirements/considerations for the system:
+
+1. **Availability** - the system must guarantee continuous read and write access to its users, lest it fails to allocate resources according to the ever-changing reality of value-differences. Such strong guarantees need not apply to *every* piece of the stack (i.e. peripheral infrastructure), but it must apply to the parts of the stack that are critical for continuous global operation. For exampl ...
```
> However, it is *not* ok for any piece of infrastructure that affects core functionality for all users to ever go down.

To me, "ever go down" means suffer a ddos attack. This does not cover the case of Alice being selectively censored like when she is unable to access her own credit history. I'd prefer wording that better captures both the prevention of ddos attacks and selective censorship.

zhekson1 committed:
added meat

added meat to end of first big section
reorged start of the 2nd (limitations) section
